File: development/java/Rubberduck.Parsing/Grammar/src/main/java/com/rubberduckvba/rubberduck/parsing/grammar/VBABaseLexer.java
package com.rubberduckvba.rubberduck.parsing.grammar;

import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.Lexer;

public abstract class VBABaseLexer extends Lexer {
    public VBABaseLexer() {
        super();
    }

    public VBABaseLexer(CharStream input) {
        super(input);
    }

    protected final int CharAtRelativePosition(int i) {
        return this._input.LA(i);
    }
	
    protected final boolean IsChar(int actual, char expected) {
        return (char)actual == expected;
    }

    protected final boolean IsChar(int actual, char... expectedOptions) {
        char actualAsChar = (char)actual;
        for (char expected : expectedOptions) {
            if (actualAsChar == expected) {
                return true;
            }
        }
        return false;
    }
}

File: Rubberduck.Parsing/Grammar/VBABaseLexer.cs
using Antlr4.Runtime;

namespace Rubberduck.Parsing.Grammar
{
    public abstract class VBABaseLexer : Lexer
    {
        public VBABaseLexer(ICharStream input) : base(input) { }

        #region Semantic predicate helper methods
        protected int CharAtRelativePosition(int i)
        {
            return _input.La(i);
        }

        protected bool IsChar(int actual, char expected)
        {
            return (char)actual == expected;
        }

        protected bool IsChar(int actual, params char[] expectedOptions)
        {
            char actualAsChar = (char)actual;
            foreach (char expected in expectedOptions)
            {
                if (actualAsChar == expected)
                {
                    return true;
                }
            }
            return false;
        }
        #endregion
    }
}
